### PR TITLE
feat: support rules_js 1.6.5 with improved version mirroring

### PR DESCRIPTION
**Pull request by gregmagolan**
Depends on https://github.com/aspect-build/rules_js/pull/576 landing first.

Introduces new npm tool mirror that vendors only the generated pnpm-lock.yaml file and not the generated defs.bzl and repositories.bzl from npm_translate_lock. The old way was no longer trivial with the introduction of aliases in the npm_translate_lock repository for bzlmod support.

The new way comes at the cost of an additional load & function call in a user's WORKSPACE, but it is simpler and more portable as generated starlark from npm_translate_lock is no longer baked but rather dynamic based on the rules_js version being used.

### CODE REVIEW

File: jasmine/dependencies.bzl [hunk 1]
```diff
@@ -14,36 +13,28 @@ load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
...
 def rules_jasmine_dependencies():
-    # The minimal version of bazel_skylib we require
-    maybe(
-        http_archive,
+    http_archive(
         name = "bazel_skylib",
-        sha256 = "c6966ec828da198c5d9adbaa94c05e3a1c7f21bd012a0b29ba8ddbccb2c93b0d",
-        urls = [
-            "https://github.com/bazelbuild/bazel-skylib/releases/download/1.1.1/bazel-skylib-1.1.1.tar.gz",
-            "https://mirror.bazel.build/github.com/bazelbuild/bazel-skylib/releases/download/1.1.1/bazel-skylib-1.1.1.tar.gz",
```
Comment (jbedard) [inline jasmine/dependencies.bzl:24]: Was there any reason to have a mirror before?

Comment (gregmagolan) [reply jasmine/dependencies.bzl:24]: The bazel build mirror urls is just copy pasted around. githbu.com is sufficient and adding the bazel mirror as a fallback doesn't add much value imo.

**Review by thesayyn — APPROVED** [review]
LGTM. I really like that we don't have to check in the generated bzl file.

File: jasmine/dependencies.bzl [hunk 2]
```diff
@@ -14,36 +13,28 @@ load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
...
-    maybe(
-        http_archive,
+    http_archive(
         name = "aspect_rules_js",
-        sha256 = "db9f446752fe4100320cf8487e8fd476b9af0adf6b99b601bcfd70b289bb0598",
-        strip_prefix = "rules_js-1.1.2",
-        url = "https://github.com/aspect-build/rules_js/archive/refs/tags/v1.1.2.tar.gz",
+        sha256 = "57b1e50da38c3b58c4fb970c3f665bd2b9705158b38a3fe9d38d7da267925c11",
+        strip_prefix = "rules_js-a47780ac9253344cc32b28c22c6da570fc05f190",
+        url = "https://github.com/aspect-build/rules_js/archive/a47780ac9253344cc32b28c22c6da570fc05f190.tar.gz",
```
Comment (jbedard) [inline jasmine/dependencies.bzl:33]: Lets wait for the release before merging though?

Comment (gregmagolan) [reply jasmine/dependencies.bzl:33]: Yup

**Review by jbedard — APPROVED** [review]

**Pull request merged**